libraries/azure_data_lake_analytics_resources.rb

```diff
@@ -0,0 +1,95 @@
+    #
+    # The dynamic part that has to be created for this resource:
+    #   Microsoft.Compute/virtualMachines?api-version=2019-12-01
+    #
+    # Parameters acquired from environment variables:
+    #   - {subscriptionId} => Required parameter. It will be acquired by the backend from environment variables.
+    #
+    # For parameters applicable to all resources, see project's README.
+    #
+    # User supplied parameters:
+    #   - resource_group => Optional parameter.
+    #   - api_version => Optional parameter. The latest version will be used unless provided.
+    #
+    #   **`resource_group`  will be used in the backend appropriately.
+    #     We don't have to do anything here.
+    #
+    # Following resource parameters have to be defined/created here.
+    #   resource_provider => Microsoft.Compute/virtualMachines
+    #     The `specific_resource_constraint` method will validate the user input
+    #       not to accept a different `resource_provider`.
+    #
+    opts[:resource_provider] = specific_resource_constraint('Microsoft.DataLakeAnalytics/accounts', opts)
+
+    # static_resource parameter must be true for setting the resource_provider in the backend.
+    super(opts, true)
+
+    # Check if the resource is failed.
+    # It is recommended to check that after every usage of superclass methods or API calls.
+    return if failed_resource?
+
+    # Define the column and field names for FilterTable.
+    #   - column: It is defined as an instance method, callable on the resource, and present `field` values in a list.
+    #   - field: It has to be identical with the `key` names in @table items that will be presented in the FilterTable.
+    # @see https://github.com/inspec/inspec/blob/master/docs/dev/filtertable-usage.md
+    table_schema = [
+      { column: :names, field: :name },
+      { column: :types, field: :type },
+      { column: :ids, field: :id },
+      { column: :provisioning_states, field: :provisioningState },
+    ]
+
+    # FilterTable is populated at the very end due to being an expensive operation.
+    AzureGenericResources.populate_filter_table(:table, table_schema)
+  end
+
+  def to_s
+    super(AzureDataLakeAnalyticsResources)
+  end
+
+  private
+
+  # Populate the @table with the resource attributes.
+  # @table has been declared in the super class as an empty array.
+  # Each item in the @table
+  #   - should be a Hash object
+  #   - should have the exact key names defined in the @table_schema as `field`.
+  def populate_table
+    # If @resources empty than @table should stay as an empty array as declared in superclass.
+    # This will ensure constructing resource and passing `should_not exist` test.
+    return [] if @resources.empty?
+
+    @resources.each do |resource|
+      @table << {
+        id: resource[:id],
+        name: resource[:name],
+        type: resource[:type],
+        provisioningState: resource[:properties][:provisioningState],
```

Comment:
why not other properties?

Reply:
This resource is incomplete Since it has dependency on **azurerm_data_lake_store** .After creation of data lake storage only, other properties will be populated .
So till then it will not go for merge.
Even terraform config also not complete .We will have to keep this PR on hold.